orttraining/orttraining/training_ops/cuda/tensor/gather_grad_impl.cu

```diff
@@ -32,10 +32,19 @@ namespace gather_grad_internal {
 // For example, given:
 //   dX_indices_sorted = [1, 1, 2, 2, 2, 3]
 //   dY_indices_sorted = [1, 4, 0, 3, 5, 2]
-// The segments will be:  '--'  '-----'  '
```

Comment:
I was expecting some duplicated code to be removed in this file?

Reply:
`Gather` nodes are replaced by `GatherInternal` only for `ORTModule`. For `ORTTrainer` the `Gather` node is not replaced. Depending on which node got executed (`GatherInternal` or `Gather`) the code path will either be the new one (the one with the precomputation) or the old one (one where all data is computed inside `GatherGrad`). Which is why I did not remove any old code from this file.